webServer.js

```diff
@@ -146,9 +154,29 @@ function start() {
     return fullContent;
   };
 
-  var setCache = function (res, hours) {
-    res.setHeader('Cache-Control', 'public, max-age=' + 60 * 60 * hours + ', immutable');
-    res.setHeader('Expires', new Date(Date.now() + 60 * 60 * hours * 1000).toUTCString());
+  var cacheCheck = function (req, res, hours, lastModified) {
+    res.setHeader("Cache-Control", "public, max-age=" + 60 * 60 * hours + ", immutable");
+    res.setHeader("Expires", new Date(Date.now() + 60 * 60 * hours * 1000).toUTCString());
+
+    //Get the if-modified-since header from the request
+    var reqModDate = req.headers["if-modified-since"];
+
+    //check if if-modified-since header is the same as the mtime of the file
+    if (lastModified && reqModDate != null) {
+      reqModDate = new Date(reqModDate);
+
+      //compared dates to the nearest second as that is what the if modified since header is accurate to
+      if (Math.floor(reqModDate.getTime() / 1000) == Math.floor(lastModified.getTime() / 1000)) {
```

Comment:
Should we use strict comparison here, or is normal equals (as it is) fine?

Reply:
Changed to triple equals.